### PR TITLE
Avoid responsibility of parsing command from string

### PR DESCRIPTION
**Pull request by future-highway**
Via @dmah42 
>parsing commands from strings is really hard.  you have to deal with a bunch of escaping and stuff that you really don't want to deal with.
today, the runtime just splits on ' ', which is going to work fine until it goes horribly wrong.

### CODE REVIEW

File: examples/cells.ts [hunk 1]
```diff
@@ -49,7 +49,8 @@ helpers.print(allocated)
 let started = await cells.start(<runtime.StartCellRequest>{
     cellName: "sleeper-cell",
     executables: [runtime.Executable.fromPartial({
-        command: "sleep 42",
+        command: "sleep",
+        commandArgs: ["42"],
```
[the remark this answers is not in the document]
Comment (future-highway) [reply examples/cells.ts:53]: Will update

File: auraed/src/runtime/mod.rs [hunk 2]
```diff
@@ -123,6 +123,7 @@ impl CellService {
             let ValidatedExecutable {
                 name: executable_name,
                 mut command,
+                command_args: _,
```
[the remark this answers is not in the document]
Comment (future-highway) [reply auraed/src/runtime/mod.rs:126]: Can't (or I don't know how to) link to comment in the code in the PR, so copying here:

From auared/src/runtime/validation.rs
// NOTE: I'm (future-highway) using this post-validation hook to add the args to the Command.
//  Usually, I prefer a 1 to 1 mapping of the unvalidated to validated struct. Adding the args
//    to the command here kind of breaks that expectation.
//  A consequence is that now we have a Command with the args already set,
//    but also a field on the validated struct with the args still there.
//  That seems wierd. We could have `command_args` always be empty (the macro
//    currently doesn't have a way to not output all the fields in the input struct)
//    but that also seems wierd.
//  This problem mainly stems from how I chose to design the `Executable` message in the proto
//   (i.e., separate fields for the program and args). This problem could be avoided by having
//   a single field (`repeated string command`), but that seemed like a worse api.

[the remark this answers is not in the document]
Comment (dmah42) [reply auraed/src/runtime/mod.rs:126]: why do we want to add the args to the command?  i would also prefer validation to not mutate things (i mentioned this before in the discord: i think these validators should be strictly non-mutating but there's no way (i know of) to enforce that in the type system).

[the remark this answers is not in the document]
Comment (future-highway) [reply auraed/src/runtime/mod.rs:126]: We don't have to. I'll undo it here and do it in CellService instead

**Review by dmah42 — APPROVED** [review]

File: auraed/src/runtime/mod.rs [hunk 3]
```diff
@@ -123,18 +123,20 @@ impl CellService {
...
                 "CellService: start() cell_name={} executable_name={} command={:?}",
                 cell_name, executable_name, command
             );
 
             // Run 'pre_exec' hooks from the context of the soon-to-be launched child.
-            let command = {
+            let _ = {
                 let executable_name_clone = executable_name.clone();
                 unsafe {
                     command.pre_exec(move || {
```
[the remark this answers is not in the document]
Comment (future-highway) [reply auraed/src/runtime/mod.rs:142]: `command.args(args)` doesn't look like it consumes `args` so it is possible without even needing a clone, but I don't think we should preemptively pass it (`pre_exec` is only a info log for now, which could also probably use `cell_name`).

**Pull request merged**